dev/NavigationView/NavigationView.cpp

```diff
@@ -1398,6 +1434,88 @@ void NavigationView::UpdateAdaptiveLayout(double width, bool forceSetDisplayMode
+            {
+                if (const auto& footerItemsRepeater = m_leftNavFooterMenuRepeater.get())
+                {
+                    const auto height = footerItemsRepeater.ActualHeight();
+                    const auto desired = footerItemsRepeater.DesiredSize().Height;
+
+                    // We know the actual height of footer items, so use that to determine how to split pane.
+                    if (const auto& menuItems = m_leftNavRepeater.get())
+                    {
+                        const auto footersActualHeight = footerItemsRepeater.ActualHeight();
+                        const auto menuItemsActualHeight = menuItems.ActualHeight();
+                        if (availableHeight > menuItemsActualHeight + footersActualHeight)
+                        {
+                            // We have enough space for two so let everyone get as much as they need.
+                            footerItemsScrollViewer.MaxHeight(footersActualHeight);
+                            availableHeight -= footersActualHeight;
+                        }
+                        else if (menuItemsActualHeight <= availableHalf)
+                        {
+                            // Footer items exceed over the half, so let's limit them.
+                            footerItemsScrollViewer.MaxHeight(availableHeight - menuItemsActualHeight);
+                            availableHeight = menuItemsActualHeight;
+                        }
+                        else if (footersActualHeight <= availableHalf)
+                        {
+                            // Menu items exceed over the half, so let's limit them.
+                            footerItemsScrollViewer.MaxHeight(footersActualHeight);
+                            availableHeight -= footersActualHeight;
+                        }
+                        else
+                        {
+                            // Both are more than half the height, so split evenly.
+                            footerItemsScrollViewer.MaxHeight(availableHalf);
+                            availableHeight = availableHalf;
+                        }
+                    }
+                    else
+                    {
+                        // Couldn't determine the menuItems.
+                        // Let's just take all the height and let the other repeater deal with it.
+                        availableHeight -= footerItemsRepeater.ActualHeight();
+                    }
+                }
+                else
+                {
+                    // No footer repeater found, do a 50/50 split.
+                    availableHeight = availableHalf;
+                }
+            }
+            if (const auto& menuItemsScrollViewer = m_menuItemsScrollViewer.get())
+            {
+                // Only update if we would actually
+                menuItemsScrollViewer.MaxHeight(availableHeight);
```

Comment:
>MaxHeight [](start = 38, length = 9)

Rather than setting the MaxHeight property on the scrollViewers do you think it is possible to do this by putting them in a grid and changing the row definitions height? then you could use Auto or *  height definitions to achieve the desired layout

Reply:
I think I tried that yesterday (getting everything right took a bit of time), and it failed for some reason. Unfortunately, I can't remember what exactly the issue was.